### PR TITLE
add initial mdbook-quiz config with sample questions

### PR DESCRIPTION
**Pull request by kwkr**
#658 
- Add `mdbook-quiz` preprocessor
- Add sample questions for each available question type

@enitrat Please have a look if you want some setting from [here](https://github.com/cognitive-engineering-lab/mdbook-quiz?tab=readme-ov-file#quiz-configuration) to be added.

- - -
This change is [Reviewable](https://reviewable.io/reviews/cairo-book/cairo-book/660)

### CODE REVIEW

**Comment by kwkr** [comment]
Hello @enitrat ! I think I got the basic setup here. Could you let me know what should be the initial scope for releasing this? Should we start with just a few questions or come up with more? I can offer my help when it comes to creating them but given my still limited experience with Cairo, somebody would definitely need to redact them.

**Review by enitrat — COMMENTED** [review]
looks good :) is this the final version or are you still working on something in this pr?

**Comment by kwkr** [comment]
I think this is all that I wanted to include in the first implementation. I'm not sure if the configuration that I included for the GH actions is enough for this so please double-check me on this.

File: quizzes/ch01-01-installation.toml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[[questions]]
+id = "1b7c7edd-8c2c-418b-ad80-c0b37b8ae215"
+type = "MultipleChoice"
+prompt.prompt = "What is Scarb?"
+answer.answer = "Cairo's package manager"
+prompt.distractors = [
+  "Rust compiler",
```
Comment (enitrat) [inline quizzes/ch01-01-installation.toml:7]: I'd rather not reference another language here. An alternative could be "a VScode extension"

File: quizzes/ch01-02-hello-world.toml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+[[questions]]
+id = "aac32d9f-b5a4-4946-81f2-eb6bcfc090ed"
+type = "MultipleChoice"
+prompt.prompt = "What is the name of the initial function that a Cairo program runs?"
+answer.answer = "main"
```
Comment (enitrat) [inline quizzes/ch01-02-hello-world.toml:5]: can be of type ShortAnswer I think

File: quizzes/ch01-02-hello-world.toml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+[[questions]]
+id = "a89b37cf-aa75-4fc7-b433-3b57d273ce1d"
+type = "ShortAnswer"
+prompt.prompt = "Which command will run the code in your project?"
+answer.answer = "scarb cairo-run"
+context = "`scarb cairo-run` will first compile and then run your code."
+
+[[questions]]
```
Comment (enitrat) [inline quizzes/ch01-02-hello-world.toml:19]: I would say too soon for this as we havent introduced variables yet

File: quizzes/ch02-01-variables-and-mutability.toml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+"""
+prompt.distractors = [
+"""
+```rust
+const MINUTES_PER_HOUR = 60;
+```
+""",
+"""
+```rust
+let y = "{}";
```
Comment (enitrat) [inline quizzes/ch02-01-variables-and-mutability.toml:18]:
```suggestion
let y = 10;
```

**Review by enitrat — COMMENTED** [review]
looks good overall. If you re interested in writing more quizzes, i would recommend using the existing quizzes from the rust book directly for the first chapters

**Comment by kwkr** [comment]
Thanks for the feedback! I will take care of it during the weekend.

**Comment by enitrat** [comment]
great! If you add more quizzes you can do it in another PR

**Review by enitrat — APPROVED** [review]
LGTM! If you want to do other quizzes, let's adapt those of the Rust book first.

**Pull request merged**